### PR TITLE
feat: modal component

### PR DESCRIPTION
**Pull request by lizaBukia**

### CODE REVIEW

File: app/Components/Button/interfaces/button-props.interface.ts [hunk 1]
```diff
@@ -4,8 +4,9 @@ import { ButtonTypeEnum } from '../enums/button-type.enum';
 
 export interface ButtonPropsInterface {
   icon?: ReactNode;
-  children: ReactNode;
+  children?: ReactNode;
```
Comment (pailodze) [inline app/Components/Button/interfaces/button-props.interface.ts:7]: why button children is optional

File: app/Components/ModalComponent/ModalComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import Input from '../Input/Input';
+import { ModalComponentPropsInterface } from './interfaces/modal-component-props.interface';
+import { ModalComponentType } from './types/modal-component.type';
+import { InputTypeEnum } from '../Input/enum/input-type.enum';
+import Button from '../Button/Button';
+import { ButtonTypeEnum } from '../Button/enums/button-type.enum';
+import { isDarkState } from '@/app/States/States';
+import { useRecoilValue } from 'recoil';
+
+const ModalComponent: ModalComponentType = (
```
Comment (pailodze) [inline app/Components/ModalComponent/ModalComponent.tsx:11]: name it modal

Comment (pailodze) [inline app/Components/ModalComponent/ModalComponent.tsx:11]: also this modal is hard coded and it should not be hardcoded

File: app/Components/ModalComponent/interfaces/modal-component-props.interface.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface ModalComponentPropsInterface {
+    Heading: string;
```
Comment (pailodze) [inline app/Components/ModalComponent/interfaces/modal-component-props.interface.ts:2]: keys should be lowercase

**Review by pailodze — CHANGES_REQUESTED** [review]

File: app/Components/ModalComponent/ModalComponent.tsx (hunk 2 above)
Comment (lizaBukia) [reply app/Components/ModalComponent/ModalComponent.tsx:11]: I don;t get it

File: app/Components/Input/Input.tsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { InputType } from './types/input.type';
 
 const Input: InputType = (props: InputPropsInterface) => {
   const isDark: boolean = useRecoilValue(isDarkState);
+  console.log(isDark)
```
Comment (pailodze) [inline app/Components/Input/Input.tsx:10]: remove console log

File: app/Components/Modal/interfaces/modal-props.interface.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export interface ModalPropsInterface {
+    isOpen: boolean;
+    children: React.ReactNode;
+    setIsOpen: (a: boolean) => void;
+
```
Comment (pailodze) [inline app/Components/Modal/interfaces/modal-props.interface.ts:5]: remove this space

File: app/Components/PlaylistModal/PlaylistModal.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import { PlaylistModalType } from './types/playlist-modal.type';
+import { isDarkState } from '@/app/States/States';
+
+const PlaylistModal: PlaylistModalType = (
+  props: PlaylistModalPropsInterface,
+) => {
+  const isDark: boolean = useRecoilValue(isDarkState);
+  const className: string = isDark ? styles.dark : styles.light;
+  return (
+    <div className={`${className} ${styles.modalWrapper} ${isDark ? styles.dark : styles.light}`}>
```
Comment (pailodze) [inline app/Components/PlaylistModal/PlaylistModal.tsx:17]: remove this ternarty

File: app/Hooks/usePlayer/usePlayer.ts [hunk 7]
```diff
@@ -4,11 +4,12 @@ import { usePlayerType } from './types/use-player.type';
 export const usePlayer = (): usePlayerType => {
   const playerRef: RefObject<HTMLAudioElement> = useRef<HTMLAudioElement>(null);
 
-  const togglePlay = (): void => {
-    if (playerRef.current?.paused) {
-      playerRef.current.play();
+  const togglePlay = () => {
```
Comment (pailodze) [inline app/Hooks/usePlayer/usePlayer.ts:7]: add this void back

**Review by pailodze — CHANGES_REQUESTED** [review]

**Pull request merged**